fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/FieldValue.java

```diff
@@ -141,7 +144,29 @@ public <M extends Message> Object eval(@Nonnull final FDBRecordStoreBase<M> stor
         // If the last step in the field path is an array that is also nullable, then we need to unwrap the value
         // wrapper.
         //
-        return NullableArrayTypeUtils.unwrapIfArray(fieldValue, getResultType());
+        return wrapPrimitive(getResultType(), NullableArrayTypeUtils.unwrapIfArray(fieldValue, getResultType()));
+    }
+
+    @Nullable
+    private static Object wrapPrimitive(@Nonnull Type type, @Nullable Object fieldValue) {
```

Comment:
Yeah, I'm not sure either `wrap` or `unwrap` are quite right...it's almost more of a "cast" primitive. But I've switched to unwrap so that it's at least consistent with `NullableArrayTypeUtils.unwrapIfArray`